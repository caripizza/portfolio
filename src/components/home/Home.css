.Home {
  .welcomeSect {
    margin-bottom: 1em;
  }

  .welcomeParagraphs {
    padding: 0.5em 1.7em 0.2em 1.7em;
    margin: 0.3em 0;
  }

  .highlight {
    &::after {
      content: "";
      display: block;
      width: 0;
      height: 2px;
      background: cyan;
      transition: width 1.3s;
    }
    &:hover::after {
      width: 100%;
    }
  }

  section {
    padding-top: 1em;
    display: inline-block;
    line-height: 1.5em;
  }

  h1,
  h2 {
    padding: 0.5em 2em 0 0.7em;
    font-style: italic;
    font-weight: 300;
    margin: 0.4em 0 0.1em 0;
  }

  h4 {
    font-weight: 700;
    color: black;
    margin: 0;
    padding: 0;
    display: inline;
  }

  img {
    margin: 0.5em 2.75em 0 1em;
    height: auto;
    float: left;
    border-radius: 50%;
    -webkit-shape-outside: circle();
    shape-outside: circle();
  }

  ul {
    list-style: square;
    color: cyan;
  }

  li span {
    color: black;
  }

  .education ul {
    list-style: none;
    line-height: 1.5em;
  }

  .detail {
    list-style: square;
  }

  li {
    animation: slide-in-left 1s linear forwards;
  }

  @keyframes slide-in-left {
    from {
      margin-left: -50px;
    }
    to {
      margin-left: 0;
    }
  }

  .half {
    display: flex;
    padding-top: 0;
  }

  .skills {
    width: 100vw;
    padding: 1em;
    border-top: 2px solid red;
  }

  .education {
    width: 100vw;
    padding: 1em;
    border-top: 2px solid gold;
  }

  .languages {
    width: 100vw;
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    min-height: 6vh;
    border: 2px solid cyan;
  }

  .languages p {
    padding: 0;
    margin: 0 0 1em 0;
    font-size: 1em;
    font-weight: 400;
  }

  .slide {
    width: 15vw;
    animation-duration: 2s;
    animation-name: slidein;
  }

  @keyframes slidein {
    from {
      margin-left: 3%;
      width: 10%;
    }
    to {
      margin-left: 1em;
      width: 15vw;
    }
  }

  .imgDiv {
    margin-left: 1em;
    min-width: 15vw;

    .logo {
      /*min-height: 5vh;*/
      /*min-width: 5vw;*/
    }
  }

  .pWelcome1 {
    padding: 0 1.7em;
    margin-top: 0;
  }

  .fadeIn {
    animation: fadein 1s;
    -moz-animation: fadein 1s; /* Firefox */
    -webkit-animation: fadein 1s; /* Safari and Chrome */
  }

  @keyframes fadein {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-moz-keyframes fadein {
    /* Firefox */
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-webkit-keyframes fadein {
    /* Safari and Chrome */
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }

  @media only screen and (max-width: 600px) and (min-width: 200px) {
    .half {
      display: block;
    }
  }

  @media screen and (min-width: 960px) {
    img {
      max-width: 9em;
    }
  }
}
