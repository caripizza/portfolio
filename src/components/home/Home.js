import React, { useState, useEffect } from 'react';
import logo from '../../../assets/caripizza_tr.png';
import styles from './Home.css';
import preloader from '../../../assets/spinnerLoad.gif';

const welcomeMsg1 =
  'Welcome! I\'m Cari, a full stack software developer, musician, and graphic designer.';
const welcomeMsg2 =
  'I started dabbling in freelance website design and digital marketing while performing and teaching music. I later made my way up to the Pacific Northwest, working in technical support and QA testing, where I fell in love with the software development cycle and decided to take the plunge into coding.';
const welcomeMsg3 =
  'In early 2019, I completed a 6 month intensive training program in Full Stack JavaScript. I made this site with React, PostCSS, and Jest, and set-up CI/CD pipelines using Travis CI and Netlify. I\'m a self taught multi-instrumentalist and audio engineer, fierce problem solver and collaborative team player, and I love learning new technologies. I\'m currently exploring AWS, Ruby, Rails, GraphQL and Angular.';

export default function Home() {
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    setTimeout(() => setLoading(false), 300);
  }, []);
  return (
    <main className={styles.Home}>
      <section className={styles.welcomeSect}>
        <div className={`${styles.imgDiv} ${styles.slide}`}>
          <img src={loading ? preloader : logo} alt="logo" className={`${styles.logo} ${styles.fadeIn} ${styles.slide}`}/>
        </div>
        <p
          className={`${styles.highlight} ${styles.pWelcome1} ${styles.fadeIn}`}
        >
          {welcomeMsg1}
        </p>
        <p
          className={`${styles.highlight} ${styles.welcomeParagraphs} ${
            styles.fadeIn
          }`}
        >
          {welcomeMsg2}
        </p>
        <p
          className={`${styles.highlight} ${styles.welcomeParagraphs} ${
            styles.fadeIn
          }`}
        >
          {welcomeMsg3}
        </p>
      </section>
      <section className={`${styles.half} ${styles.fadeIn}`}>
        {/* <article className={styles.skills}> */}
        <article className={styles.skills}>
          <h2 className={styles.highlight}>Expertise</h2>
          <ul>
            <li>
              <span>JavaScript (ES6+), DOM manipulation</span>
            </li>
            <li>
              <span>UI/UX Mobile-first/Responsive design principles</span>
            </li>
            <li>
              <span>
                BDD/TDD: Selenium, Jenkins, Jasmine, Jest, Enzyme, Mocha/Chai,
                Protractor
              </span>
            </li>
            <li>
              <span>
                Frameworks/Tools: Node.js, Express.js, Vue.js, React, Redux,
                jQuery, AJAX, Axios, Postman, SASS, PostCSS, Styled-Components,
                Material UI, Angular
              </span>
            </li>
            <li>
              <span>
                Database: Google Cloud Platform, Firebase/Firestore, Cloud
                Functions, MongoDB, Mongoose, PostgreSQL, MySQL
              </span>
            </li>
            <li>
              <span>CD/CI: Git, GitHub, Travis CI, Gitlab</span>
            </li>
            <li>
              <span>
                A/V + Design: ProTools, Cubase, Logic Pro, Final Cut Pro, Adobe
                Photoshop, Illustrator, InDesign, AfterEffects
              </span>
            </li>
            <li>
              <span>
                CDN: Squarespace, Wordpress / Ecommerce: Stripe, PayPal,
                American Express, Vantiv, Authorize.Net
              </span>
            </li>
            <li>
              <span>
                Admin/SEO: G Suite, Sheets, Docs, Analytics, Ads, Search
                Console, Facebook Analytics
              </span>
            </li>
          </ul>
        </article>
        <article className={styles.education}>
          <h2 className={styles.highlight}>Education</h2>
          <ul>
            <li>
              <span>
                <strong>
                  Advanced Software Development in Full Stack JavaScript
                </strong>
                <br />
                Alchemy Code Lab (Portland, 2018-2019)
              </span>
            </li>
            <li className={styles.detail}>
              <span>
                MERN-stack professional training in vanilla JavaScript (ES6+),
                DOM manipulation, RESTful APIs, Node.js apps with Express,
                Vue.js, PostgreSQL, React, Redux, MongoDB, Mongoose
              </span>
              <br />
            </li>
            <li>
              <br />
              <span>
                <strong>Bachelor of Fine Arts</strong>
                <br />
                The University of Texas at Austin (2002-2006)
              </span>
            </li>
            <li className={styles.detail}>
              <span>
                Studio Art major with concentrations in Video, Drafting, Painting, and Printmaking
              </span>
            </li>
            <br />
            <li>
              <span>
                <strong>
                  Courses
                </strong>
              </span>
            </li>
            <li className={styles.detail}>
              <span>
                Intro to Website Development
                <br />
                Portland Community College (2015)
              </span>
            </li>
            <li className={styles.detail}>
              <span>
                Seo Strategy & Training
                <br />
                The City of Austin Small Business Development Program (2013)
              </span>
            </li>
          </ul>
        </article>
      </section>
      <section className={styles.languages}>
        <p>JavaScript</p>
        <p>HTML/CSS</p>
        <p>SQL</p>
        <p>Ruby/Rails</p>
      </section>
    </main>
  );
}
