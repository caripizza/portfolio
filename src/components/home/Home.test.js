import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Home', () => {
  it('matches a snapshot', () => {
    const component = renderer.create(
      <Home />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
