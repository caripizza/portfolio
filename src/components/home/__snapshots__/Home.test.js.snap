// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches a snapshot 1`] = `
<main
  className="Home"
>
  <section>
    <img
      alt="logo"
      className="slide"
      src=""
    />
    <p>
      
  Welcome! I'm Cari, a Full Stack JavaScript developer, musician, and graphic designer.

    </p>
    <p>
      
  After a decade performing and teaching music, I started dabbling in freelance website design and digital marketing for small businesses.
  I made my way up to the Pacific Northwest and joined the tech scene as a support agent, later working as a QA tester for a small software company.
  While testing all weekly releases and pairing with UI/UX to propose new system features, I fell in love with the software development cycle and decided to take the plunge into coding.

    </p>
    <p>
      
  In early 2019, I completed a 6 month intensive training program in full stack JavaScript, focusing in front-end website design as well as DevOps, managing the release flow.
  I made this site with React and styled it with PostCSS. I'm a self taught multi-instrumentalist and audio engineer, fierce problem solver and collaborative team player, and I love learning new technologies.

    </p>
  </section>
  <section
    className="half"
  >
    <article
      className="skills"
    >
      <h2>
        <strong>
          Skills
        </strong>
      </h2>
      <ul>
        <li>
          <span>
            <strong>
              Design:
            </strong>
             UI/UX, Mobile-first principles
          </span>
        </li>
        <li>
          <span>
            <strong>
              BDD:
            </strong>
             Selenium, Jenkins, Jasmine
          </span>
        </li>
        <li>
          <span>
            <strong>
              TDD:
            </strong>
             Jest, Enzyme, Mocha/Chai, React
          </span>
        </li>
        <li>
          <span>
            <strong>
              Frameworks:
            </strong>
             Node, Express, Vue, React, Redux, jQuery, Sass, PostCSS, Styled-Components
          </span>
        </li>
        <li>
          <span>
            <strong>
              Databases:
            </strong>
             Firebase/Firestore, MongoDB, Mongoose, PostgreSQL
          </span>
        </li>
        <li>
          <span>
            <strong>
              DevTools:
            </strong>
             Cloud Functions, Travis CI, DNS, Heroku, Netlify, Git, Slack, Trello, JIRA, Atlassian, GitHub, npm, pip
          </span>
        </li>
        <li>
          <span>
            <strong>
              Audio/Video:
            </strong>
             ProTools, Cubase, Logic Pro, Final Cut Pro
          </span>
        </li>
        <li>
          <span>
            <strong>
              Adobe CC:
            </strong>
             Photoshop, Illustrator, InDesign, AfterEffects
          </span>
        </li>
        <li>
          <span>
            <strong>
              CMS:
            </strong>
             Squarespace, Wordpress
          </span>
        </li>
        <li>
          <span>
            <strong>
              CDN:
            </strong>
             Cloudflare/AWS
          </span>
        </li>
        <li>
          <span>
            <strong>
              Admin:
            </strong>
             G Suite (Sheets, Docs, Slides, Analytics, Ads)
          </span>
        </li>
        <li>
          <span>
            <strong>
              SEO:
            </strong>
             Google Search Console, Ads, Analytics
          </span>
        </li>
      </ul>
    </article>
    <article
      className="education"
    >
      <h2>
        <strong>
          Academic Background
        </strong>
      </h2>
      <ul>
        <li>
          <span>
            <strong>
              Advanced Software Development in Full Stack JavaScript:
            </strong>
            <br />
            Alchemy Code Lab (Portland OR) | 2018-2019
          </span>
        </li>
        <li
          className="detail"
        >
          <span>
            MERN-stack professional training with emphasis on clean code, team-building, individual problem solving, mob programming + pair programming, and TDD testing
          </span>
        </li>
        <li
          className="detail"
        >
          <span>
            3 part program with focus on vanilla JavaScript, SPA app development with RESTful APIs, full-stack Node.js apps made with Express, Vue.js, PostgreSQL, React, Redux, MongoDB + Mongoose
          </span>
          <br />
        </li>
        <li>
          <br />
          <span>
            <strong>
              Bachelor of Fine Arts:
            </strong>
            <br />
            The University of Texas at Austin | 2002-2006
          </span>
        </li>
        <li
          className="detail"
        >
          <span>
            Major: Studio Art | Concentrations: Video, Drafting, Painting, Printmaking
          </span>
        </li>
        <li
          className="detail"
        >
          <span>
            Graduated with University Honors, Cum Laude
          </span>
        </li>
      </ul>
    </article>
  </section>
  <section
    className="languages"
  >
    <p>
      JavaScript
    </p>
    <p>
      HTML/HTML5
    </p>
    <p>
      CSS/CSS3
    </p>
    <p>
      SQL
    </p>
    <p>
      Python
    </p>
  </section>
</main>
`;
